added remove-dependencies support to win_chocolatey

MrBones757 opened the pull request:
Related to issue https://github.com/chocolatey/chocolatey-ansible/issues/13

vexx32 commented:
@MrBones757 CI failure looks to be a sanity test; you'll need to add the new parameter to the module documentation as well. 🙂

MrBones757 commented:
how silly of me! fixed now :)

vexx32 commented:
Hmm, that's odd. 

```
ERROR: Found 1 shebang issue(s) which need to be resolved:
ERROR: plugins/modules/win_chocolatey.py:0:0: module should not be executable
See documentation for help: https://docs.ansible.com/ansible/2.9/dev_guide/testing/sanity/shebang.html
```

You didn't change the executable bit of the .py file, did you? 🤔

MrBones757 commented:
The only thing i changed in the py file was i added an example and the parameter documentation - i confirmed this with the git changes just in case and compared to some other windows modules.... seems very strange....

jborean93 commented on chocolatey/plugins/modules/win_chocolatey.py:
```diff
@@ -80,6 +80,12 @@
+    - Remove a package's dependencies on uninstall.
+    type: bool
+    default: no
+    version_added: '2.10'
```
This should reflect the collection version that the feature was added and not the Ansible version.

MrBones757 replied:
Should this be changed to 1.0.3 to align with the next release?

jborean93 commented:
The sanity tests are correct, the diff is showing the execute bit is now set `644` -> `755`

![image](https://user-images.githubusercontent.com/8462645/82500605-05bcfe80-9b37-11ea-9e81-2cac10008b38.png)

Maybe your git config is configured to set the executable bit for certain extensions? If you edited this from WSL that could have also changed the permissions due to the NTFS driver that handles all that.

vexx32 commented:
@MrBones757 are you able to correct the executable bit issue so that we can get this merged? 🙂

MrBones757 commented:
Hello!

Thanks for reminding me about this. Lost track of it with everything going on at work.
This was indeed edited with WSL and I should have thought about this as its happened before -.-.

Should be resolved now if im correct.

vexx32 commented:
Thanks! :blush:

Hmm, it looks like Github didn't register a change. Did you push the changes?

MrBones757 commented:
should be good now - apologies

vexx32 commented:
No worries, and thanks! 😊 💖 

I'll let CI finish and merge this in the morning if everything checks out. 🙂

MrBones757 commented:
the only other thing that @jborean93 pointed out was the version_added. I have it set to 2.10 but he suggested in the comments above that it should be the collection version - can you confirm this

vexx32 commented:
Yeah, that's correct. Probably many of the existing version_added references are going to be not quite correct since these modules were originally part of ansible's core packages, we'll need to go through and set them to the correct versions before we push the next release.

For the purposes of this one in particular, though, we'll want to set it to `1.1.0` 🙂

MrBones757 commented:
version 1.1.0 added instead of 2.10

vexx32 commented on chocolatey/plugins/modules/win_chocolatey.py:
```diff
@@ -99,7 +105,7 @@
       to current package passed arguments.
     type: bool
-    version_added: '2.10'
+    version_added: '1.1.0'
```
Looks like you changed the wrong version here? The one you initially added was on line 88 🙂
